modules/lang-painless/src/test/java/org/elasticsearch/painless/BasicExpressionTests.java

```diff
@@ -135,4 +137,73 @@ public void testPrecedence() {
         assertEquals(2, exec("int x = 5; return (x+x)/x;"));
         assertEquals(true, exec("boolean t = true, f = false; return t && (f || t);"));
     }
+
+    public void testNullSafeDeref() {
+        // Objects in general
+        assertNull(                      exec("String a = null; return a?.toString()"));   // Call
+        assertNull(                      exec("String a = null; return a?.length()"));     // Call and box
+        assertEquals("foo",              exec("String a = 'foo'; return a?.toString()"));  // Call
+        assertEquals(Integer.valueOf(3), exec("String a = 'foo'; return a?.length()"));    // Call and box
```

Comment:
I swear I'm missing something.  I thought that top-level variables wouldn't be working with this based on the grammar changes I see...